src/s1reader/s1_orbit.py

```diff
@@ -337,7 +399,267 @@ def get_orbit_file_from_list(zip_path: str, orbit_file_list: list) -> str:
+        https://sentinel.esa.int/web/sentinel/user-guides/sentinel-1-sar/naming-conventions
+    orbit_file_list : list
+        List of the orbit files that exists in the system.
+
+    Returns:
+    --------
+    orbit_file : str
+        Path to the orbit file, or `None` if no orbit file was found.
+    '''
+    # check the existence of input file path and directory
+    if not os.path.exists(zip_path):
+        raise FileNotFoundError(f"{zip_path} does not exist")
+
+    # extract platform id, start and end times from swath file name
+    mission_id, t_swath_start_stop = get_file_name_tokens(zip_path)
+
+    # choose the RESORB file only
+    resorb_file_list = [orbit_file for orbit_file in orbit_file_list
+                        if '_RESORB_' in os.path.basename(orbit_file)]
+    pad_1min = datetime.timedelta(seconds=PADDING_SHORT)
+    resorb_filename_earlier = None
+    resorb_filename_later = None
+    for resorb_file in resorb_file_list:
+        if mission_id not in os.path.basename(resorb_file):
+            continue
+
+        # 1. Try to find the orbit file that covers
+        # sensing time - margin_start_time
+        t_swath_start_stop_safe = [t_swath_start_stop[0] - pad_1min,
+                                   t_swath_start_stop[1] + pad_1min]
+        if _covers_timeframe(resorb_file, t_swath_start_stop_safe) and \
+            resorb_filename_later is None:
+            print('Found RESOEB file covering the S1 SAFE frame.')
+            resorb_filename_later = resorb_file
+            continue
+
+        # 2. Try to find the orbit file that covers the sensing start-stop
+        # with small padding (like 60 sec.)
+        t_swath_start_stop_anx = [t_swath_start_stop[0] - margin_start_time,
+                                  t_swath_start_stop[0] - margin_start_time + 2*pad_1min]
+        if _covers_timeframe(resorb_file, t_swath_start_stop_anx) and \
+            resorb_filename_earlier is None:
+            print('Found RESOEB file covering ANX before sensing start')
+            resorb_filename_earlier = resorb_file
+            continue
+
+        # break out of the for loop when the RESORB files are found
+        if resorb_filename_earlier and resorb_filename_later:
+            break
+
+    # if 1. and 2. are successful, then try to concatenate them
+    if resorb_filename_earlier and resorb_filename_later:
+        if concatenate_resorb:
+            # BE CAREFUL ABOUT THE ORDER HOW THEY ARE CONCATENATED.
+            # See NOTE in retrieve_orbit_file() for detail.
+            concat_resorb_filename = combine_xml_orbit_elements(resorb_filename_later,
+                                                                resorb_filename_earlier)
+            print('RESORB Concatenation successful.')
+            return concat_resorb_filename
+        else:
+            return [resorb_filename_earlier, resorb_filename_later]
+
+    print('Cannot find single RESORB file that meets the time frame criteria.')
+    return None
```

Comment:
```suggestion
    return
```
Not needed. It will "return" `None` regardless of you adding `None` or not